src/model/migrations.js

```diff
@@ -51,20 +58,76 @@ export default async function runMigrations() {
           id,
           imported: false,
           name: DEFAULT_WALLET_NAME,
+          primary: true,
           type: WalletTypes.mnemonic,
         };
 
         const wallets = { [id]: currentWallet };
 
-        store.dispatch(walletsUpdate(wallets));
-        store.dispatch(walletsSetSelected(currentWallet));
+        await store.dispatch(walletsUpdate(wallets));
+        await store.dispatch(walletsSetSelected(currentWallet));
       }
     }
   };
 
   migrations.push(v1);
 
-  /***** Migration v1 ends here  *****/
+  /*
+   *************** Migration v2 ******************
+   * This step handles the addition of "primary wallets"
+   * which are the only wallets allowed to create new accounts under it
+   */
+  const v2 = async () => {
+    const { wallets, selected } = store.getState().wallets;
+
+    // Check if we have a primary wallet
+    const primaryWallet = Object.keys(wallets).find(key => {
+      const wallet = wallets[key];
+      return wallet.primary;
+    });
+
+    // If there's no primary wallet, we need to find
+    // if there's a wallet with seed phrase that wasn't imported
+    // and set it as primary
+    if (!primaryWallet) {
+      let primaryWalletKey = null;
+      Object.keys(wallets).some(key => {
+        const wallet = wallets[key];
+        if (wallet.type === WalletTypes.mnemonic && !wallet.imported) {
+          primaryWalletKey = key;
+          return true;
+        }
+        return false;
+      });
+
+      // If there's no wallet with seed phrase that wasn't imported
+      // let's find a wallet with seed phrase that was imported
+      if (!primaryWalletKey) {
+        Object.keys(wallets).some(key => {
+          const wallet = wallets[key];
+          if (wallet.type === WalletTypes.mnemonic) {
+            primaryWalletKey = key;
+            return true;
+          }
+          return false;
+        });
+      }
+
+      if (primaryWalletKey) {
+        const updatedWallets = { ...wallets };
+        updatedWallets[primaryWalletKey].primary = true;
```

Comment:
Given `...wallets` is a shallow copy of the redux wallets object, the object at `updatedWallets[primaryWalletKey]` is still the same object and will complain about immutability when you try to set `.primary = true` on it. May need to do something like `updatedWallets[primaryWalletKey] = { ...updatedWallets[primaryWalletKey], primary: true }`